# 14.75 problem set 1 - Max Ghenis

When written answers are required, they are provided below the question in **bold.**

In [28]:
library(dplyr, quietly=T)
library(ggplot2, quietly=T)
library(AER, quietly=T)

## Question 1

Create data frame with 100 observations with

a) A variable `t` running from `1` to `100`

b) `alpha` which has value `3` for all 100 observations

c) Another variable `e_t`, a normal variable with mean 0 and standard deviation 1

d) `x_t` a random uniform variable over [0, 1]

In [2]:
SIZE = 100
set.seed(0)

In [3]:
df = data.frame(t=1:SIZE,
                alpha=3,
                e_t=rnorm(100),
                x_t=runif(100))

e) An outcome variable `y_t` where `beta=2`, `y_t` = `alpha + beta * x_t + e_t`.

In [4]:
BETA = 2
df$y_t = df$alpha + BETA * df$x_t + df$e_t

f) Estimate `beta_hat`

**1.83**

In [5]:
m = lm(y_t ~ x_t, data=df)
summary(m)


Call:
lm(formula = y_t ~ x_t, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.23533 -0.60614 -0.06678  0.56760  2.50565 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   3.0983     0.1656  18.712  < 2e-16 ***
x_t           1.8257     0.3223   5.665 1.48e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.8858 on 98 degrees of freedom
Multiple R-squared:  0.2467,	Adjusted R-squared:  0.239 
F-statistic: 32.09 on 1 and 98 DF,  p-value: 1.476e-07


f) i. Test H0: `beta = 0`

**p-value of 1.48e-7 < 0.05, so we can reject `beta=0`.**

f) ii. Test H0: `beta = 1.2`

**95% CI [1.19, 2.47] includes 1.2, so we cannot reject `beta=1.2`.**

In [6]:
confint(m)

2.5 %    97.5 %  
(Intercept) 2.769741 3.426900
x_t         1.186110 2.465289

g) Create `v_t` as a random normal variable with mean 0. Generate `q_t` as `q_t = x_t + 2*x_t^3 + v_t`.

In [7]:
df$v_t = rnorm(SIZE)

In [8]:
df$q_t = df$x_t + 2 * df$x_t ^ 3 + df$v_t

What is the correlation between q and x?

**0.64**

In [9]:
cor(df$q_t, df$x_t)

[1] 0.6420868

Generate outcome variable `z_t` where `beta=2`, `gamma=3`, `z_t = alpha + beta * x_t + gamma * q_t + e_t`.

In [10]:
GAMMA = 3

In [11]:
df$z_t = df$alpha + BETA * df$x_t + GAMMA * df$q_t + df$e_t

Estimate `beta_hat` from (misspecified) model:
`z_t = alpha + beta * x_t + u_t`

**11.4**

In [12]:
m = lm(z_t ~ x_t, data=df)
summary(m)


Call:
lm(formula = z_t ~ x_t, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-8.6316 -2.4583  0.2057  2.1837  7.5694 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   1.6213     0.6186   2.621   0.0102 *  
x_t          11.4267     1.2041   9.490 1.55e-15 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 3.309 on 98 degrees of freedom
Multiple R-squared:  0.4789,	Adjusted R-squared:  0.4736 
F-statistic: 90.06 on 1 and 98 DF,  p-value: 1.548e-15


Test H0: `beta = 2`.

**Estimated `beta` = 11.43. 95% CI = [9.04, 13.82] so we can reject `beta=2` under this misspecified model.**

In [13]:
confint(m)

2.5 %    97.5 %   
(Intercept) 0.393792  2.848848
x_t         9.037274 13.816108

Discuss.

**This is the result of omitted variable bias: `z_t` is being affected by `q_t` which is correlated with `x_t` in a nonlinear way that `x_t` can't soak up alone (if it were linear we'd expect `beta_hat = beta + gamma = 5`, rather than 11.43).**

## Question 2

Use the `AssassinationsData.RData` dataset.

In [14]:
load("AssassinationsData.RData")

(a) Regress whether the institution changed from 1 year before the attempt to 1 year
after the attempt [`absnpolity2dummy11`] on whether or not the attempt was successful
[`success`]. As in the original paper, make sure this is conditional on there being a “serious attempt” [`seriousattempt`].

In [19]:
serious = AssassinationsData[AssassinationsData$seriousattempt == 1,]

In [25]:
m = lm(absnpolity2dummy11 ~ success, data=serious, na.action=na.omit)
summary(m)


Call:
lm(formula = absnpolity2dummy11 ~ success, data = serious, na.action = na.omit)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.15254 -0.04348 -0.04348 -0.04348  0.95652 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)   
(Intercept)  0.04348    0.02020   2.152  0.03247 * 
success      0.10906    0.03901   2.796  0.00564 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.2563 on 218 degrees of freedom
  (12 observations deleted due to missingness)
Multiple R-squared:  0.03462,	Adjusted R-squared:  0.03019 
F-statistic: 7.817 on 1 and 218 DF,  p-value: 0.005637


i. Interpret βb.

**When an assassination is seriously attempted, successes are associated with an additional 10.9 percent chance that institutions will change from the year before the attempt to the year after.**

ii. What assumptions do we need to interpret this as the expected difference in outcomes between when the attempt succeeds and when it fails?

**answer**

iii. Why might these assumptions fail?

**answer**

iv. Test the hypothesis that β = 0 at the 5% level. Do you reject or fail to reject the
null?

**The p-value associated with `success` is 0.00564 < 0.05, so we can reject the null that B=0.**

(b) Control for fixed effects of the type of weapon used [weapondum2, weapondum3, ...] in
the specification from (a).

In [ ]:
# Regress with controls

i. Why would you want to include these controls?

**Answer**

ii. Does your interpretation of βb change? If so, how so?

**Answer**

iii. Under what assumptions can we interpret βb as the expected difference in outcomes
between when the attempt succeeds and when it fails?

**Answer**

iv. Test the hypothesis that β = 0.1 at the 5% level. Do you reject or fail to reject the
null?

**Answer**

## Question 3: Instrumental Variables

Use the AJRData.RData dataset.

In [27]:
load('AJRData.RData')

(a) Regress the log GDP per capita in 1995 [logpgp95] on the average protection against
expropriation risk [avexpr]. In this and other problems, restrict the sample to that in
the paper [baseco should equal 1].
R Tip: For plotting, consider the package ggplot2.

In [30]:
ajr = AJRData[AJRData$baseco == 1, ]

In [31]:
m = lm(logpgp95 ~ avexpr, data=ajr)
summary(m)


Call:
lm(formula = logpgp95 ~ avexpr, data = ajr)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.8715 -0.4644  0.1683  0.4610  1.1413 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.66038    0.40851  11.408  < 2e-16 ***
avexpr       0.52211    0.06119   8.533 4.72e-12 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.7132 on 62 degrees of freedom
  (99 observations deleted due to missingness)
Multiple R-squared:  0.5401,	Adjusted R-squared:  0.5327 
F-statistic: 72.82 on 1 and 62 DF,  p-value: 4.724e-12


i. Interpret βols.

**answer**

ii. Is your interpretation causal? Why or why not?

**answer**

iii. Plot log GDP per capita in 95 against avexpr.

In [ ]:
# Plot

(b) Regress the average protection against expropriation risk on settler mortality [logem4].
Call this `pi_hat`.

In [32]:
m = lm(logem4 ~ avexpr, data=ajr)
summary(m)


Call:
lm(formula = logem4 ~ avexpr, data = ajr)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.7613 -0.7403 -0.0394  0.2941  3.4182 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  7.55772    0.62049   12.18  < 2e-16 ***
avexpr      -0.44519    0.09294   -4.79 1.08e-05 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.083 on 62 degrees of freedom
  (99 observations deleted due to missingness)
Multiple R-squared:  0.2701,	Adjusted R-squared:  0.2584 
F-statistic: 22.95 on 1 and 62 DF,  p-value: 1.077e-05


In [40]:
pi_hat = coef(m)['avexpr']
pi_hat

avexpr 
-0.4451896

i. Plot avexpr against settler mortality.

In [ ]:
# Plot

ii. Interpret the relationship.
(c) Regress the log GDP per capita in 1995 on settler mortality. Call this `gamma_hat`.

In [41]:
m = lm(logpgp95 ~ logem4, data=ajr)
summary(m)


Call:
lm(formula = logpgp95 ~ logem4, data = ajr)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.7545 -0.5386  0.1412  0.4607  1.4059 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 10.73057    0.36718  29.224  < 2e-16 ***
logem4      -0.57297    0.07616  -7.523 2.66e-10 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.7604 on 62 degrees of freedom
  (99 observations deleted due to missingness)
Multiple R-squared:  0.4772,	Adjusted R-squared:  0.4688 
F-statistic:  56.6 on 1 and 62 DF,  p-value: 2.659e-10


In [42]:
gamma_hat = coef(m)['logem4']
gamma_hat

logem4 
-0.5729682

i. Plot the relationship.

In [ ]:
# Plot

ii. Compute `gamma_hat/pi_hat`.

In [43]:
gamma_hat / pi_hat

logem4 
1.28702

(d) Compute a 2SLS regression of log GDP per capita in 1995 on avexpr, using settler
mortality as an instrumental variable.
R Tip: For IV, try the “ivreg” command in the “AER” package.

In [ ]:
# Regress

i. Interpret βb2sls.

**answer**

ii. Compare βb2sls to βbols. Are they similar or different? Why do you think this is the
case?

**answer**

iii. Under what assumption can you interpret each of these causally.

**answer**

iv. Compare βb2sls to bγ/bπ. Explain, mathematically, the relationship between βb2sls, πb, and γb.

**answer**